src/main/java/io/strimzi/kafka/bridge/KafkaBridgeConsumer.java

```diff
@@ -151,6 +155,10 @@ public void assign(List<SinkTopicSubscription> topicSubscriptions) {
             topicPartitions.add(new TopicPartition(topicSubscription.getTopic(), topicSubscription.getPartition()));
         }
 
+        if (topicPartitions.isEmpty()) {
+            this.unsubscribe();
+        }
```

Comment:
same as described before. Run the unsubscribe here, not returning and then the code runs the assign.
Just running the assign with an empty topics list should work.

Reply:
Thanks for the review, exactly, will update the code!